server/index.js

```diff
@@ -155,21 +155,83 @@ submissionStreams.on('connection', socket => {
   if (!keywords || !subreddits || toggleInbox === null) {
     throw new ClientError(400, 'missing search terms');
   }
+
   const connectedAt = Date.now() / 1000;
 
   const subStream = createSearchStream(socket.user.requester, subreddits);
 
   const parsedKw = parseKeywords(keywords);
 
+  const botRequester = toggleInbox === 'true'
+    ? new Snoowrap({
+      userAgent: 'keyword finder bot v1.0 by (/u/buddhababy23)',
+      clientId: process.env.SCRIPT_ID,
+      clientSecret: process.env.SCRIPT_SECRET,
+      username: process.env.REDDIT_USER,
+      password: process.env.REDDIT_PW
+    })
+    : null;
+
   subStream.on('item', submission => {
     if (connectedAt > submission.created_utc) return;
     if (parsedKw.some(word => submission.title.toLowerCase().includes(word.toLowerCase()))) {
       socket.emit('new_submission', submission);
+      if (botRequester !== null) {
+        socket.user.requester.getMe()
+          .then(user => {
+            botRequester.composeMessage({
+              to: user,
+              subject: 'Keyword Found!',
+              text: submission.url
+            });
+          })
+          .catch(err => console.error(err));
+      }
     }
   });
 
+  const userId = socket.user.userId;
+
   socket.on('disconnect', socket => {
     subStream.end();
+
+    if (botRequester) {
+      const sql = `
+            select *
+              from "subscriptions"
+            where "userId" = $1;
+      `;
+
+      const params = [userId];
+
+      db.query(sql, params)
+        .then(result => {
+          const [subscription] = result.rows;
+
+          const sql = subscription
+            ? `
+                update "subscriptions"
+                  set "keywords" = $1,
+                      "subreddits" = $2
+                where "userId" = $3
+            returning *;`
+            : `
+        insert into "subscriptions" ("keywords", "subreddits", "userId")
+                             values ($1, $2, $3)
+                          returning *;`;
+
+          const params = [keywords, subreddits, userId];
```

Comment:
ok, here you should do what's called an "upsert" (insert or update). 

if users are only allowed to have one subscription at a time (a reasonable limitation for mvp). You need to add a unique constraint to your `"userId"` in `"subscriptions"`. 

Then you won't need to query the database and then conditionally insert or update.

```sql
insert into "subscriptions" ("keywords", "subreddits", "userId")
values ($1, $2, $3)
on conflict ("userId")
do update
set "keywords" = $1,
    "subreddits" = $2
```